Add indexer migrations and sqlc models

jarrel-b opened the pull request:
**Changes**
Adds migrations and sqlc models for the Indexer in preparation for the deep refresh PR. Instead of the `sqlc` package, there will be a `coregen` and `indexergen` packages (also open to better names) to namespace the backend models from the indexer models.

The new directory structure under `db` now will look something like:
```
db
├── migrate.go
├── migrations
│   ├── core # migrations for the backend
│   │   ├── 000001_create_core_tables.down.sql
│   │   ├── 000001_create_core_tables.up.sql
│   │   └── ...
│   └── indexer # migrations for the indexer
│   │   ├── 000001_create_core_tables.down.sql
│   │   ├── 000001_create_core_tables.up.sql
│   │   └── ...
├── queries
│   ├── core.sql # queries for the backend
│   └── indexer.sql # queries for the indexer
└── sqlc
    ├── coregen # code generated for the backend
    └── indexergen # core generated for the indexer
```

radazen commented:
Thanks for splitting this stuff up! Happy to merge it into my branch. Thoughts/comments/questions:

- I've been splitting queries into multiple .sql files, one per Storage API type, so `queries/core` will actually be a directory
- What if we renamed the `sqlc` directory to `gen`? Technically both the queries and the generated stuff are all sqlc (since the queries are annotated with sqlc conventions and use sqlc parameter syntax), so imho it's a good time to drop the `sqlc` entirely.
- On that topic, I've also renamed the `sqlc` package to `db` on my branch. I picked `sqlc` originally to make it obvious which types were hand-written `persist` structs and which were generated by sqlc, but at the point where _all_ of our database I/O structs are being generated by sqlc, I don't think we need to call the package `sqlc` anymore
- For generated package names: what about `gen/coredb` and `gen/indexerdb`? I'll probably still set up `db` as an import alias, but if I don't, `coredb.User` is descriptive and concise.

So with those edits, maybe something like this?

```
db
├── migrate.go
├── migrations
│   ├── core # migrations for the backend
│   │   ├── 000001_create_core_tables.down.sql
│   │   ├── 000001_create_core_tables.up.sql
│   │   └── ...
│   └── indexer # migrations for the indexer
│   │   ├── 000001_create_core_tables.down.sql
│   │   ├── 000001_create_core_tables.up.sql
│   │   └── ...
├── queries
│   ├── core/ # queries for the backend
│   │   ├── user.sql
│   │   ├── gallery.sql
│   │   └── ...
│   └── indexer.sql # queries for the indexer
└── gen
    ├── coredb # code generated for the backend
    └── indexerdb # code generated for the indexer
```

jarrel-b commented:
> Thanks for splitting this stuff up! Happy to merge it into my branch. Thoughts/comments/questions:
> 
> * I've been splitting queries into multiple .sql files, one per Storage API type, so `queries/core` will actually be a directory
> * What if we renamed the `sqlc` directory to `gen`? Technically both the queries and the generated stuff are all sqlc (since the queries are annotated with sqlc conventions and use sqlc parameter syntax), so imho it's a good time to drop the `sqlc` entirely.
> * On that topic, I've also renamed the `sqlc` package to `db` on my branch. I picked `sqlc` originally to make it obvious which types were hand-written `persist` structs and which were generated by sqlc, but at the point where _all_ of our database I/O structs are being generated by sqlc, I don't think we need to call the package `sqlc` anymore
> * For generated package names: what about `gen/coredb` and `gen/indexerdb`? I'll probably still set up `db` as an import alias, but if I don't, `coredb.User` is descriptive and concise.
> 
> So with those edits, maybe something like this?
> 
> ```
> db
> ├── migrate.go
> ├── migrations
> │   ├── core # migrations for the backend
> │   │   ├── 000001_create_core_tables.down.sql
> │   │   ├── 000001_create_core_tables.up.sql
> │   │   └── ...
> │   └── indexer # migrations for the indexer
> │   │   ├── 000001_create_core_tables.down.sql
> │   │   ├── 000001_create_core_tables.up.sql
> │   │   └── ...
> ├── queries
> │   ├── core/ # queries for the backend
> │   │   ├── user.sql
> │   │   ├── gallery.sql
> │   │   └── ...
> │   └── indexer.sql # queries for the indexer
> └── gen
>     ├── coredb # code generated for the backend
>     └── indexerdb # code generated for the indexer
> ```

Way better names! Made those suggested changes

radazen reviewed (approved):
🙌🙌🙌
